apis/externalsecrets/v1beta1/secretstore_conjur_types.go

```diff
@@ -0,0 +1,24 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+// ConjurProvider configures an store to sync secrets using a Conjur backend.
+type ConjurProvider struct {
+	ServiceURL         *string `json:"serviceUrl,omitempty"`
+	ServiceUser        *string `json:"serviceUser,omitempty"`
+	ServiceAPIKey      *string `json:"serviceApiKey,omitempty"`
```

Comment:
The API key is a sensitive value and should be sourced from a secret resource. This is a pattern that is implemented throughout all the providers.

We have a [esmeta.SecretStoreSelector](https://github.com/external-secrets/external-secrets/blob/d605a5b450abb8072090ef115396f8b618aec2ec/apis/meta/v1/types.go#L19-L30) type that should be used for this case.

Note: `Namespace` must only be taken into account when using a ClusterSecretStore ([see here for example](https://github.com/external-secrets/external-secrets/blob/6de9a138db833cbe03da57682fff34117e94f175/pkg/provider/gcp/secretmanager/auth.go#L58-L65)). When a ClusterSecretStore is used then the `namespace` is mandatory.

A namespaced `SecretStore` must lookup the secret from the same namespace as the SecretStore resource itself. The `namespace` must not be set in this case.

Suggestion:
```yaml
apiVersion: external-secrets.io/v1beta1
kind: ClusterSecretStore
metadata:
  name: conjur-example
spec:
  provider:
    conjur:
      serviceUrl: "http://localhost:8080"
      serviceUser: "Dave@BotApp"
      serviceApiKey:
        secretRef:
          name: "conjur-creds"
          namespace: "external-secrets"
          key: "apikey"
```

Reply:
This is so true, can't believe I missed this one 😓 big +1

(to the other comments as well)

Reply:
I was wondering how we could do this! Thank you for this information! I'm looking into it now. ❤️